akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -39,20 +39,37 @@ private class BackoffOnRestartSupervisor(
         // directive match {
         case Restart ⇒
           val childRef = sender()
-          become({
-            case Terminated(`childRef`) ⇒
-              unbecome()
-              child = None
-              val restartDelay = BackoffSupervisor.calculateDelay(restartCount, minBackoff, maxBackoff, randomFactor)
-              context.system.scheduler.scheduleOnce(restartDelay, self, BackoffSupervisor.StartChild)
-              restartCount += 1
-            case _ ⇒ // ignore
-          }, discardOld = false)
+          become(waitChildTerminatedBeforeBackoff(childRef) orElse handleBackoff)
           Stop
         case other ⇒ other
       }
   }
 
+  def waitChildTerminatedBeforeBackoff(childRef: ActorRef): Receive = {
+    case Terminated(`childRef`) ⇒
+      become(receive)
+      child = None
+      val restartDelay = BackoffSupervisor.calculateDelay(restartCount, minBackoff, maxBackoff, randomFactor)
+      context.system.scheduler.scheduleOnce(restartDelay, self, BackoffSupervisor.StartChild)
+      restartCount += 1
+
+    case StartChild ⇒
+      become(stashedStartChild(childRef) orElse handleBackoff)
```

Comment:
When would we get this `StartChild`? When it terminates above, it goes into `receive` before scheduling it.